source/schemas/types.txt

```diff
@@ -509,3 +509,124 @@ with the ``title`` field set to the embedded object's type name:
             }
       }
    }
+
+.. _geospatial:
+
+Geospatial Data
+---------------
+
+Geospatial data describes points and other data on the earth's surface. The 
+Realm SDKs support the creation of numerous types of geospatial data. However, 
+only persisted GeoJSON Point data can be queried. 
+
+GeoJSON Point
+~~~~~~~~~~~~~
+
+A GeoJSON Point (GeoPoint) is a single location on the Earth's surface. The GeoPoint 
+schema must contain the following fields:
+
+- ``type`` (string), which is always set to "Point"
+
+- ``coordinates`` (array of doubles). The ``coordinates`` array must contain  
+  at least 2 double values, and may contain a third:
+    
+    - The first double is the longitude of the point, between -180 and 180.
+    - The second is the latitude, between -90 and 90.
+    - The optional third value represents the elevation/altitude of the point, 
+      in meters.
+
+To store a GeoPoint in Atlas for later querying with :ref:`Atlas Device Sync <sync>`, 
+the following rules also apply:
+
+- It must be of type ``object``
+- It must be embedded within another type
+- Additional properties are allowed within the ``coordinates`` array, as long as 
+  they are doubles.
+- Additional properties in the GeoPoint object are also allowed.
+ 
+The following is the Device Sync schema of an object with an embedded GeoPoint 
+property named "location":
+
+.. code-block:: json
+   :copyable: false
+
+   {
+     "title": "MyObject",
+     "properties": {
+       "location": {
+         "bsonType": "object",
+         "required": [ "type" ],
+         "properties": {
+           "type": {
+             "bsonType": "string",
+           },
+           "coordinates": {
+             "bsonType": "array",
+             "items": {
+               "bsonType": "double"
+             }
+           }
+          }
+         }
+       },
+    "additionalProperties": true
+   }
+
+The following code block shows a document that follows the schema. It has a 
+single embedded GeoPoint object named "location". Both the containing object 
+and the GeoPoint object contain an extra property.
+
+.. code-block:: json
+   :copyable: false
+   :emphasize-lines: 4-9
+
+   {
```

Comment:
[nit] probably worth adding some whitespace here (similar to the spacing in the above code block) to make the code block a bit more readable